qiskit/transpiler/preset_passmanagers/__init__.py

```diff
@@ -213,6 +214,13 @@ def generate_preset_pass_manager(
             stacklevel=2,
         )
 
+    if (
+        coupling_map
+        and isinstance(coupling_map, list)
+        and all(isinstance(sublist, list) for sublist in coupling_map)
+    ):
+        coupling_map = CouplingMap(coupling_map)
+
```

Comment:
we can probably do
```suggestion
    if coupling is not None and not isinstance(coupling_map, CouplingMap):
        coupling_map = CouplingMap(coupling_map)
```
because the `CouplingMap` constructor will already have to validate its input, which would just be duplicated and liable to get out-of-sync with this check.

Reply:
Good point, thanks!